infra/aws/terraform/prow-build-cluster/modules/iam/policies.tf

```diff
@@ -0,0 +1,171 @@
+      "eks:TagResource",
+      "iam:AttachRolePolicy",
+      "iam:CreateOpenIDConnectProvider",
+      "iam:CreatePolicy",
+      "iam:CreateRole",
+      "iam:CreateServiceLinkedRole",
+      "iam:GetOpenIDConnectProvider",
+      "iam:GetPolicy",
+      "iam:GetPolicyVersion",
+      "iam:GetRole",
+      "iam:GetRolePolicy",
+      "iam:GetUser",
+      "iam:ListAttachedRolePolicies",
+      "iam:ListRolePolicies",
+      "iam:PassRole",
+      "iam:PutRolePolicy",
+      "iam:TagOpenIDConnectProvider",
+      "iam:TagPolicy",
+      "iam:TagRole",
+      "kms:CreateAlias",
+      "kms:CreateGrant",
+      "kms:CreateKey",
+      "kms:DescribeKey",
+      "kms:EnableKeyRotation",
+      "kms:GetKeyPolicy",
+      "kms:GetKeyRotationStatus",
+      "kms:ListAliases",
+      "kms:ListResourceTags",
+      "kms:PutKeyPolicy",
+      "kms:TagResource",
+      "logs:CreateLogGroup",
+      "logs:DescribeLogGroups",
+      "logs:ListTagsLogGroup",
+      "logs:PutRetentionPolicy",
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:PutObject",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "prow_cluster_destroy" {
+  statement {
+    sid       = ""
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "ec2:DeleteEgressOnlyInternetGateway",
+      "ec2:DeleteInternetGateway",
+      "ec2:DeleteLaunchTemplate",
+      "ec2:DeleteNatGateway",
+      "ec2:DeleteRoute",
+      "ec2:DeleteRouteTable",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteSubnet",
+      "ec2:DeleteVpc",
+      "ec2:DescribeInternetGateways",
+      "ec2:DescribeNatGateways",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DescribeRouteTables",
+      "ec2:DescribeSecurityGroups",
+      "ec2:DescribeVpcs",
+      "ec2:DetachInternetGateway",
+      "ec2:DisassociateAddress",
+      "ec2:DisassociateRouteTable",
+      "ec2:DisassociateVpcCidrBlock",
+      "ec2:ReleaseAddress",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RevokeSecurityGroupIngress",
+      "eks:DeleteAddon",
+      "eks:DeleteCluster",
+      "eks:DeleteNodegroup",
+      "eks:DescribeAddon",
+      "eks:DescribeCluster",
+      "eks:DescribeNodegroup",
+      "iam:DeleteOpenIDConnectProvider",
+      "iam:DeletePolicy",
+      "iam:DeleteRole",
+      "iam:DeleteRolePolicy",
+      "iam:DetachRolePolicy",
+      "iam:ListAttachedRolePolicies",
+      "iam:ListInstanceProfilesForRole",
+      "iam:ListPolicyVersions",
+      "iam:ListRolePolicies",
+      "kms:DeleteAlias",
+      "kms:DescribeKey",
+      "kms:ScheduleKeyDeletion",
+      "logs:DeleteLogGroup",
+      "s3:PutObject",
+    ]
+  }
+}
+
+resource "aws_iam_policy" "prow_cluster_maintainer" {
+  name   = "ProwClusterMaintainer"
+  path   = "/"
+  policy = data.aws_iam_policy_document.prow_cluster_maintainer.json
+}
+
+resource "aws_iam_policy" "prow_cluster_destroy" {
+  name   = "ProwClusterDestroy"
+  path   = "/"
+  policy = data.aws_iam_policy_document.prow_cluster_destroy.json
+}
```

Comment:
I was thinking about yet another policy that would allow for running terraform plan. It could be shared with contributors. WDYT @xmudrii ?

Reply:
Added ProwClusterViewer policy and in the latest commit.